# What Is the Sharpe Ratio?
The Sharpe ratio compares the return of an investment with its risk. It's a mathematical expression of the insight that excess returns over a period of time may signify more volatility and risk, rather than investing skill.

$$ SharpeRatio = \frac{R_{p}-R_{f}}{\sigma_{p}} $$

Where:
* Rp = Return of Portfolio
* Rf = Risk-free rate
* Op = Standard Deviation of the portfolio's excess returns

https://www.investopedia.com/terms/s/sharperatio.asp#:~:text=To%20calculate%20the%20Sharpe%20ratio%2C%20investors%20can%20subtract%20the%20risk,deviation%20(the%20asset's%20volatility.)

In [ ]:
import numpy as np
import pandas as pd
import yfinance as yf
import matplotlib.pyplot as plt

In [ ]:
rfr = 0.03 #Risk-Free rate

In [ ]:
russell = yf.Ticker('IWM') #iShares Russell 200 ETF
data_russell = russell.history(interval='1d', start='2016-01-01', end='2025-01-25')
print(data_russell  )

                                 Open        High         Low       Close  \
Date                                                                        
2016-01-04 00:00:00-05:00   98.255547   98.450975   96.896440   97.722565   
2016-01-05 00:00:00-05:00   98.077893   98.228903   97.313949   97.935760   
2016-01-06 00:00:00-05:00   96.558898   97.358375   95.937086   96.443420   
2016-01-07 00:00:00-05:00   94.800041   95.492919   93.707422   93.867317   
2016-01-08 00:00:00-05:00   94.044970   94.595715   92.081814   92.250587   
...                               ...         ...         ...         ...   
2025-01-17 00:00:00-05:00  226.929993  226.970001  224.669998  225.460007   
2025-01-21 00:00:00-05:00  227.520004  229.889999  226.990005  229.789993   
2025-01-22 00:00:00-05:00  229.350006  229.850006  227.830002  228.289993   
2025-01-23 00:00:00-05:00  226.929993  229.350006  226.550003  229.330002   
2025-01-24 00:00:00-05:00  228.820007  230.229996  228.039993  228.690002   

In [ ]:
#The returns are estimated as the log difference from the closing price:
russell_close = data_russell['Close']
russell_returns = np.log(russell_close).diff()
russell_returns = russell_returns.dropna()
print(russell_returns)

Date
2016-01-05 00:00:00-05:00    0.002179
2016-01-06 00:00:00-05:00   -0.015355
2016-01-07 00:00:00-05:00   -0.027074
2016-01-08 00:00:00-05:00   -0.017374
2016-01-11 00:00:00-05:00   -0.004342
                               ...   
2025-01-17 00:00:00-05:00    0.004312
2025-01-21 00:00:00-05:00    0.019023
2025-01-22 00:00:00-05:00   -0.006549
2025-01-23 00:00:00-05:00    0.004545
2025-01-24 00:00:00-05:00   -0.002795
Name: Close, Length: 2278, dtype: float64


In [ ]:
russell_mean_returns = np.mean(russell_returns) *252
russell_std_returns = np.std(russell_returns)*np.sqrt(252)
print(f"russell_mean_returns:" ,russell_mean_returns)
print(f"russell_std_returns:" ,russell_std_returns)

russell_mean_returns: 0.09405583550079015
russell_std_returns: 0.23134943493997906


In [ ]:
#Calculate the Sharpe Ratio
sharpe_ratio = (russell_mean_returns - rfr) / russell_std_returns
print(f"sharpe_ratio:" ,sharpe_ratio)

sharpe_ratio: 0.2768791526005183


In [ ]:
#Total Returns
russell_total_returns = russell_close[-1] / russell_close[0]
print(f"russell_total_returns:" ,russell_total_returns)

russell_total_returns: 2.3401964853241846


<ipython-input-13-60707240273f>:2: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  russell_total_returns = russell_close[-1] / russell_close[0]


In [21]:
print(f"\nMean Returns: {russell_mean_returns * 100:.2f}%")
print(f"Standard Deviation of Returns: {russell_std_returns*100:.2f}%")
print(f"Sharpe Ratio: {sharpe_ratio}")
print(f"Total Returns: {russell_total_returns*100:.2f}%")


Mean Returns: 9.41%
Standard Deviation of Returns: 23.13%
Sharpe Ratio: 0.2768791526005183
Total Returns: 234.02%


# iShares Russell 2000 ETF Performance Metrics
* **Mean Returns:** *9.41%* - Average expected return over the period, indicating profitable performance.
* **Standard Deviation of Returns:** *23.13%* - Reflects high volatility, suggesting returns can vary significantly from the mean.
* **Sharpe Ratio:** *0.277* - Indicates low risk-adjusted return; positive but marginal compensation for risk.
* **Total Returns:** *234.02%* - Cumulative return, representing substantial growth of the investment over the specified period.
